### PR TITLE
(feat) O3-4181 Display Patient Age and Implement View/Edit Results Functionality for Lab Technicians

### PR DESCRIPTION
**Pull request by its-kios09**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Screenshots

updates-lab

## Related Issue

see: https://openmrs.atlassian.net/browse/O3-4181

### CODE REVIEW

**Comment by pirupius** [comment]
@its-kios09 did this design get these designs from zeplin? or does this have approval from the UI/UX team?

**Comment by its-kios09** [comment]
> @its-kios09 - let's separate adding the additional patient details from UI enhancements. The UI one will likely take some time to allow for proper designs

let me revert the enhancements

**Comment by its-kios09** [comment]
@ojwanganto I've completed the update. The latest version is in the GIF video. 

updated-order

**Comment by its-kios09** [comment]
@pirupius, I have reverted the UI to the previous version due to UI/UX concerns. The current changes include the addition of patient details such as patient age and the ability to view and edit lab results.

File: src/components/orders-table/orders-data-table.component.tsx [hunk 1]
```diff
@@ -23,11 +23,12 @@ import { formatDate, parseDate, usePagination } from '@openmrs/esm-framework';
 import { Order } from '@openmrs/esm-patient-common-lib';
 import React, { useMemo, useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import { useLabOrders, useSearchGroupedResults } from '../../laboratory-resource';
+import usePatientAge, { useLabOrders, useSearchGroupedResults } from '../../laboratory-resource';
 import { FulfillerStatus, OrdersDataTableProps } from '../../types';
 import ListOrderDetails from './list-order-details.component';
 import styles from './orders-data-table.scss';
 import { OrdersDateRangePicker } from './orders-date-range-picker';
+import { PatientAge } from './patient-age.component';
```
Comment (pirupius) [inline src/components/orders-table/orders-data-table.component.tsx:31]: Let's eliminate the need for the extra hook

File: src/components/orders-table/orders-data-table.component.tsx [hunk 2]
```diff
@@ -124,11 +126,12 @@ const OrdersDataTable: React.FC<OrdersDataTableProps> = (props) => {
...
   const tableRows = useMemo(() => {
-    return paginatedLabOrders.map((patient) => ({
-      id: patient.patientId,
-      patientName: patient.orders[0].patient?.display?.split('-')[1],
-      orders: patient.orders,
-      totalOrders: patient.orders?.length,
+    return paginatedLabOrders.map((order) => ({
+      id: order.patientId,
+      patientName: order.orders[0].patient?.display?.split('-')[1],
+      age: <PatientAge patientUuid={order.patientId} />,
```
Comment (pirupius) [inline src/components/orders-table/orders-data-table.component.tsx:132]: Changes will trickle here

File: src/components/orders-table/list-order-details.scss [hunk 3]
```diff
@@ -55,3 +55,6 @@
   margin-left: layout.$spacing-05;
   margin-top: layout.$spacing-07;
 }
+.accordionContainer {
```
Comment (pirupius) [inline src/components/orders-table/list-order-details.scss:58]: Minor formatting. Let's add an empty line before a new class

File: src/components/orders-table/patient-age.component.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
Comment (pirupius) [inline src/components/orders-table/patient-age.component.tsx:1]: Ditto

**Review by pirupius — COMMENTED** [review]
Thanks @its-kios09 some minor comments and clean ups to do before we can get this in

File: src/routes.json [hunk 5]
```diff
@@ -96,6 +96,11 @@
       "name": "reject-lab-request-tests-ordered-action",
       "component": "rejectLabRequestAction",
       "slot": "tests-ordered-actions-slot"
+    },
+      {
```
[the remark this answers is not in the document]
Comment (its-kios09) [reply src/routes.json:100]: fixed

File: src/components/orders-table/list-order-details.scss (hunk 3 above)
Comment (its-kios09) [reply src/components/orders-table/list-order-details.scss:58]: sorted

**Comment by its-kios09** [comment]
@pirupius Hello Pius, I investigated where we could retrieve the patient’s age using the existing hooks. The model for /orders/ does not seem to include person properties it provides only the` uuid and display which is name`. unless you have an idea I can explore

File: src/laboratory-resource.ts [hunk 6]
```diff
@@ -88,3 +88,12 @@ export function rejectLabOrder(orderId: string, comment: string, abortController
...
 }
+
+const usePatientAge = (uuid: string) => {
+  const customRepresentation = `custom:(person:(age))`;
+  const url = `${restBaseUrl}/patient/${uuid}?v=${customRepresentation}`;
+  const { isLoading, error, data } = useSWR<FetchResponse<{ person: { age: number } }>>(url, openmrsFetch);
+  return { isLoading, error, patientAge: data?.data?.person?.age };
+};
+
+export default usePatientAge;
```
[the remark this answers is not in the document]
Comment (its-kios09) [reply src/laboratory-resource.ts:99]: @pirupius  on the existing hook the representation is set to full

File: src/components/orders-table/patient-age.component.tsx (hunk 4 above)
Comment (its-kios09) [reply src/components/orders-table/patient-age.component.tsx]: Okay sure

**Comment by its-kios09** [comment]
@pirupius I found an alternative way to retrieve the age from the existing hook by updating the v=full to a custom representation, and it is working well and optimized. however I have updated the PR
Let me know if you need any further adjustments!

File: src/laboratory-resource.ts [hunk 7]
```diff
@@ -88,3 +90,12 @@ export function rejectLabOrder(orderId: string, comment: string, abortController
     },
   });
 }
+
+const usePatientAge = (uuid: string) => {
```
Comment (pirupius) [inline src/laboratory-resource.ts:94]: Let's do some clean up and delete this since it's no longer necessary

Comment (its-kios09) [reply src/laboratory-resource.ts]: Resolved

**Review by pirupius — APPROVED** [review]

**Pull request merged**